clients/java/src/main/java/io/camunda/client/impl/search/UserTaskQueryImpl.java

```diff
@@ -0,0 +1,127 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.client.impl.search;
+
+import static io.camunda.client.api.search.SearchRequestBuilders.searchRequestPage;
+import static io.camunda.client.api.search.SearchRequestBuilders.userTaskFilter;
+import static io.camunda.client.api.search.SearchRequestBuilders.userTaskSort;
+
+import io.camunda.client.api.JsonMapper;
+import io.camunda.client.api.search.FinalSearchQueryStep;
+import io.camunda.client.api.search.SearchRequestPage;
+import io.camunda.client.api.search.UserTaskFilter;
+import io.camunda.client.api.search.UserTaskQuery;
+import io.camunda.client.api.search.UserTaskSort;
+import io.camunda.client.api.search.response.SearchQueryResponse;
+import io.camunda.client.impl.http.HttpCamundaFuture;
+import io.camunda.client.impl.http.HttpClient;
+import io.camunda.client.protocol.rest.UserTaskItem;
+import io.camunda.client.protocol.rest.UserTaskSearchQueryRequest;
+import io.camunda.client.protocol.rest.UserTaskSearchQueryResponse;
+import java.time.Duration;
+import java.util.function.Consumer;
+import org.apache.hc.client5.http.config.RequestConfig;
+
+public class UserTaskQueryImpl
+    extends TypedSearchRequestPropertyProvider<UserTaskSearchQueryRequest>
+    implements UserTaskQuery {
+
+  private final UserTaskSearchQueryRequest request;
+  private final JsonMapper jsonMapper;
+  private final HttpClient httpClient;
+  private final RequestConfig.Builder httpRequestConfig;
+
+  public UserTaskQueryImpl(final HttpClient httpClient, final JsonMapper jsonMapper) {
+    request = new UserTaskSearchQueryRequest();
+    this.jsonMapper = jsonMapper;
+    this.httpClient = httpClient;
+    httpRequestConfig = httpClient.newRequestConfig();
+  }
+
+  @Override
+  public FinalSearchQueryStep<UserTaskItem> requestTimeout(final Duration requestTimeout) {
+    return null;
```

Comment:
❌ The request timeout needs to be passed on to the HTTP client configuration, so it's respected. Have a look at the `ProcessInstanceQueryImpl`for reference.